maven-core/src/main/java/org/apache/maven/plugin/PluginParameterExpressionEvaluator.java

```diff
@@ -40,7 +40,7 @@
  * <tr><td><code>session</code></td>           <td></td>               <td>the actual {@link MavenSession}</td></tr>
  * <tr><td><code>session.*</code></td>         <td>(since Maven 3)</td><td></td></tr>
  * <tr><td><code>localRepository</code></td>   <td></td>
- *                                             <td>{@link MavenSession#getLocalRepository()}</td></tr>
+ *                                             <td>{@link MavenSession#getLocalRepository()} DEPRECATED: Avoid use of {@link org.apache.maven.artifact.repository.ArtifactRepository} type. If you need access to local repository, switch to '${repositorySystemSession}' expression and get LRM from it instead.</td></tr>
```

Comment:
I think some Plugin deleopers might have no clue what LRM is, probabbly also give a link to the wiki/jira/... that explains some details.

Reply:
Updated javadoc, added link to the issue. Issue updated with "how to fix" https://issues.apache.org/jira/browse/MNG-7706